packages/js/assets/user-feedback-form.js

```diff
@@ -0,0 +1,122 @@
+(function (window, document) {
+  // eslint-disable-next-line quotes
+  const TEMPLATE = `$$TEMPLATE$$`
+  const ENDPOINT = 'https://api.honeybadger.io/v1/feedback'
+  const OPTIONS_KEY = 'honeybadgerUserFeedbackOptions'
+
+  const HoneybadgerUserFeedbackForm = function () {};
+  HoneybadgerUserFeedbackForm.prototype.build = function () {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this
+    self.element = document.createElement('div')
+    self.element.id = 'honeybadger-feedback-wrapper'
+    self.element.innerHTML = TEMPLATE
+    self.element.onclick = function (e) {
+      if (e.target !== self.element) return;
+      self.close();
+    }
+
+    document.body.appendChild(self.element)
+
+    self.form = self.element.getElementsByTagName('form')[0]
+    self.form.addEventListener('submit', function (e) {
+      e.preventDefault()
+      self.submit()
+    })
+
+    const cancelButton = document.getElementById('honeybadger-feedback-cancel')
+    cancelButton.onclick = (e) => {
+      e.preventDefault()
+      self.close()
+    }
+    const closeButton = document.getElementById('honeybadger-feedback-close')
+    closeButton.onclick = (e) => {
+      e.preventDefault()
+      self.close()
+    }
+
+    const formOptions = self.getOptions()
+    const { messages = {}, buttons = {}, labels = {} } = formOptions
+    for (let i in messages) {
+      const element = document.getElementById(`honeybadger-feedback-${i}`)
+      if (element) {
+        element.innerText = messages[i]
+      }
+    }
+    for (let i in labels) {
+      const element = document.getElementById(`honeybadger-feedback-label-${i}`)
+      if (element) {
+        element.innerText = labels[i]
+      }
+    }
+    for (let i in buttons) {
+      const element = document.getElementById(`honeybadger-feedback-${i}`)
+      if (element) {
+        element.value = buttons[i]
+      }
+    }
+  };
+
+  HoneybadgerUserFeedbackForm.prototype.close = function () {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
```

Comment:
It's a practice I have since my early days in js scripting; maybe it's a bad practice?
The reason I still go for it is because `this` can point to a different object depending on the context (i.e. `this` inside `function () {}` vs `() => {}` (arrow function)) or even the original caller of the function.

I think I'll just disable the rule.